File: 백준/Silver/13305. 주유소/주유소.py
import sys
input = sys.stdin.readline
sys.setrecursionlimit(10**6)


N = int(input())
road = [int(x) for x in input().split()]
cost = [int(x) for x in input().split()]

min_cost = cost[0]
total = 0

for i in range(len(cost)-1):
    if min_cost > cost[i]:
        min_cost = cost[i]

    total += min_cost * road[i]

print(total)

File: 백준/Silver/13305. 주유소/주유소.js
let fs = require('fs')
let input = fs.readFileSync("/dev/stdin").toString().trim().split("\n")
// let input = fs.readFileSync("./EX.txt").toString().trim().split("\n")

const N = parseInt(input[0])
const road = input[1].split(" ").map(x => parseInt(x))
const cost = input[2].split(" ").map(x => parseInt(x))

let minCost = cost[0]
let total = minCost * road[0]

for (let i = 1; i < N-1; i++) {
    minCost = Math.min(minCost, cost[i])
    total += minCost * road[i]
}

console.log(total)